src/components/MapView/MapView.tsx

```diff
@@ -84,15 +154,27 @@ export let MapView = withStore(
       onDragEnd={(map: MapboxGL.Map) => store.set('mapCenter')(map.getCenter())}
       onZoomEnd={(map: MapboxGL.Map) => store.set('mapZoom')(map.getZoom())}
       zoom={[store.get('mapZoom')]}
+      onClick={() => removePopup(store)}
     >
       {representativePoints.length && <GeoJSONLayer
+        id='representativePoints'
         data={representativePointsToGeoJSON(adequacies)(representativePoints)}
         circlePaint={representativePointCircleStyle}
+        circleOnClick={(point: any) => store.set('representativePointClicked')(point)}
+        circleOnMouseEnter={() => store.set('mapCursor')('pointer')}
+        circleOnMouseLeave={() => store.set('mapCursor')('')}
       />}
       {providers.length && <GeoJSONLayer
+        id='providers'
         data={providersToGeoJSON(providers)}
         circlePaint={providerCircleStyle}
+        circleOnClick={(point: any) => store.set('providerClicked')(point)}
+        circleOnMouseEnter={() => store.set('mapCursor')('pointer')}
+        circleOnMouseLeave={() => store.set('mapCursor')('')}
       />}
+      {representativePointClicked && representativePointPopup(representativePointClicked)}
```

Comment:
Convention is to instantiate React Components with JSX syntax, so it's clear that it's a component. So capitalize the component name, and change this to `<RepresentativePointPopup point={representativePointClicked} />`.

Reply:
Done